src/pip/_internal/download.py

```diff
@@ -330,7 +330,7 @@ def __call__(self, req):
         # Set the url of the request to the url without any credentials
         req.url = url
 
-        if username is not None and password is not None:
+        if username is not None or password is not None:
```

Comment:
It looks like None values should be converted to an empty string before calling HTTPBasicAuth: https://github.com/pypa/pip/blob/2e51624bbb42c83ac3ec5898f71657ea5186a784/src/pip/_vendor/requests/auth.py#L28

Reply:
Looks like this pattern was used prior to keyring implementation...

https://github.com/pypa/pip/blob/e34769861ec5ed8aa5ce23670e111fe8db446c39/pip/download.py#L96